src/stemmablenote.ts

```diff
@@ -178,66 +200,60 @@ export class StemmableNote extends Note {
   }
 
   // Set the stem length to a specific. Will override the default length.
-  setStemLength(height) {
+  setStemLength(height: number): this {
     this.stem_extension_override = height - Stem.HEIGHT;
     return this;
   }
 
   // Get the top and bottom `y` values of the stem.
-  getStemExtents() {
-    return this.stem.getExtents();
+  getStemExtents(): Record<string, number> | undefined {
+    return this.stem?.getExtents();
   }
 
-  // Sets the current note's beam
-  setBeam(beam) {
-    this.beam = beam;
-    return this;
-  }
-
-  // Get the `y` value for the top/bottom modifiers at a specific `textLine`
-  getYForTopText(textLine) {
+  /** Gets the `y` value for the top modifiers at a specific `textLine`. */
+  getYForTopText(textLine: number): number {
     const extents = this.getStemExtents();
     if (this.hasStem()) {
       return Math.min(
-        this.stave.getYForTopText(textLine),
-        extents.topY - this.render_options.annotation_spacing * (textLine + 1)
+        this.stave?.getYForTopText(textLine) || 0,
+        extents?.topY || 0 - this.render_options.annotation_spacing * (textLine + 1)
       );
     } else {
-      return this.stave.getYForTopText(textLine);
+      return this.stave?.getYForTopText(textLine) || 0;
     }
   }
 
-  getYForBottomText(textLine) {
+  /** Gets the `y` value for the bottom modifiers at a specific `textLine`. */
+  getYForBottomText(textLine: number): number {
     const extents = this.getStemExtents();
     if (this.hasStem()) {
       return Math.max(
-        this.stave.getYForTopText(textLine),
-        extents.baseY + this.render_options.annotation_spacing * textLine
+        this.stave?.getYForTopText(textLine) || 0,
+        extents?.baseY || 0 + this.render_options.annotation_spacing * textLine
       );
     } else {
-      return this.stave.getYForBottomText(textLine);
+      return this.stave?.getYForBottomText(textLine) || 0;
     }
   }
 
-  hasFlag() {
-    return Flow.getGlyphProps(this.duration).flag && !this.beam;
+  hasFlag(): boolean {
+    return Flow.getGlyphProps(this.duration).flag && !this.beam && this.flag;
```

Comment:
@0xfe I find it strange that this.flag was not in the condition. I good also say that this.flag nevers get initialised. Is this a bug?

Reply:
I'm not exactly sure -- I have to dig in. I'd say add a TODO, but don't change the logic.

Reply:
Without this check stavenote is now crashing. There must be a difference that I have introduced. But the reality is that stavenote uses hasFlag to safely access the flag.

Reply:
This is once again this getGlyphProp. Could you provide a description of the aim of this function?

Reply:
Ok I will revert the change. I have seen that the condition used in `StaveNote` to see if there is a `flag` is `hasFlag`. I have replaced the condition is this class to the same check `hasFlag` and the tests pass correctly.